src/server/pps/server/api_server.go

```diff
@@ -3113,36 +3113,45 @@ func (a *apiServer) RunCron(ctx context.Context, request *pps.RunCronRequest) (r
 	if pipelineInfo.Input == nil {
 		return nil, errors.Errorf("pipeline must have a cron input")
 	}
-	if pipelineInfo.Input.Cron == nil {
-		return nil, errors.Errorf("pipeline must have a cron input")
-	}
 
-	cron := pipelineInfo.Input.Cron
+	// find any cron inputs
+	var crons []*pps.CronInput
+	pps.VisitInput(pipelineInfo.Input, func(in *pps.Input) {
+		if in.Cron != nil {
+			crons = append(crons, in.Cron)
+		}
+	})
 
-	// We need the DeleteFile and the PutFile to happen in the same commit
-	_, err = pachClient.StartCommit(cron.Repo, "master")
-	if err != nil {
-		return nil, err
+	if len(crons) < 1 {
+		return nil, errors.Errorf("pipeline must have a cron input")
 	}
-	if cron.Overwrite {
-		// get rid of any files, so the new file "overwrites" previous runs
-		err = pachClient.DeleteFile(cron.Repo, "master", "")
-		if err != nil && !isNotFoundErr(err) && !pfsServer.IsNoHeadErr(err) {
-			return nil, errors.Wrapf(err, "delete error")
+
+	// make a tick on each cron input
+	for _, cron := range crons {
+		// We need the DeleteFile and the PutFile to happen in the same commit
+		_, err = pachClient.StartCommit(cron.Repo, "master")
+		if err != nil {
+			return nil, err
+		}
+		if cron.Overwrite {
+			// get rid of any files, so the new file "overwrites" previous runs
+			err = pachClient.DeleteFile(cron.Repo, "master", "")
+			if err != nil && !isNotFoundErr(err) && !pfsServer.IsNoHeadErr(err) {
+				return nil, errors.Wrapf(err, "delete error")
+			}
 		}
-	}
 
-	// Put in an empty file named by the timestamp
-	_, err = pachClient.PutFile(cron.Repo, "master", time.Now().Format(time.RFC3339), strings.NewReader(""))
-	if err != nil {
-		return nil, errors.Wrapf(err, "put error")
-	}
+		// Put in an empty file named by the timestamp
+		_, err = pachClient.PutFile(cron.Repo, "master", time.Now().Format(time.RFC3339), strings.NewReader(""))
+		if err != nil {
+			return nil, errors.Wrapf(err, "put error")
+		}
 
-	err = pachClient.FinishCommit(cron.Repo, "master")
-	if err != nil {
-		return nil, err
+		err = pachClient.FinishCommit(cron.Repo, "master")
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Probably better to do this in an atomic commit using a `PutFileClient` so there's no risk of crashing and leaving a dangling open commit.

Reply:
Also is it possible to do the cron commits in a transaction so you don't trigger multiple times? Would be good to have a test for the multiple cron input case as well.

Reply:
PutFileClient doesn't allow putting files into different repos, so I just used a transaction.